src/engine/source/builder/builders/opBuilderHelperActiveResponse.cpp

```diff
@@ -0,0 +1,405 @@
+                }
+                else
+                {
+                    return base::result::makeFailure(event, failureTrace6);
+                }
+                isLocal = true;
+            }
+            else if (!location.compare("ALL"))
+            {
+                // TODO: This case generated one message per active agent in analysisd,
+                // currently only the "all" text will be sent, which is not supported at
+                // the moment by execd.
+                agentID = "all";
+                isAll = true;
+            }
+            else // Check if it is a number (specific agent id)
+            {
+                if (ar::isStringNumber(location))
+                {
+                    agentID = location;
+                    isID = true;
+                }
+                else
+                {
+                    return base::result::makeFailure(event, failureTrace7 + location);
+                }
+            }
+
+            if (agentID.empty())
+            {
+                return base::result::makeFailure(event, failureTrace4);
+            }
+
+            // Check existence and not emptyness of extraArgsRefValue if used
+            if (!extraArgsRefValue.empty())
+            {
+                const auto extraArgsArray = event->getArray(extraArgsRefValue);
+                if (!extraArgsArray.has_value())
+                {
+                    return base::result::makeFailure(event, failureTrace1);
+                }
+                else
+                {
+                    for (const auto arrayElement : extraArgsArray.value())
+                    {
+                        // fill "extra_args" in json with this array
+                        const auto resolvedElement {arrayElement.getString()};
+
+                        if (resolvedElement)
+                        {
+                            baseJson.appendString(
+                                std::string_view {resolvedElement.value()},
+                                std::string_view {ar::EXTRA_ARGS_PATH});
+                        }
+                        else
+                        {
+                            return base::result::makeFailure(event, failureTrace8);
+                        }
+                    }
+                }
+            }
+
+            try
+            {
+                json::Json jsonEvent {event->str().c_str()};
+                baseJson.merge(jsonEvent, std::string_view {ar::ALERT_PATH});
+            }
+            catch (const std::runtime_error& e)
+            {
+                return base::result::makeFailure(event, failureTrace5 + e.what());
+            }
+
+            const std::string query {baseJson.str()};
+
+            // Append header message
+            const std::string completeMesage =
+                fmt::format("(local_source) [] N{}{} {} {}",
+                            isLocal ? 'R' : 'N',
+                            isAll|isID ? 'S' : 'N',
```

Comment:
use logical and non-binary operators for boolean variables
```suggestion
                            (isAll || isID) ? 'S' : 'N',
```